google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java

```diff
@@ -0,0 +1,516 @@
+    dsaSetting.toBuilder().addAllFeeds(ImmutableList.of(StringValue.of(feedResourceName))).build();
+
+    // Creates the campaign object to update.
+    Campaign campaign =
+        Campaign.newBuilder()
+            .setResourceName(ResourceNames.campaign(customerId, campaignId))
+            .setDynamicSearchAdsSetting(dsaSetting)
+            .build();
+
+    // Creates the update operation and sets the update mask.
+    CampaignOperation operation =
+        CampaignOperation.newBuilder()
+            .setUpdate(campaign)
+            .setUpdateMask(FieldMasks.allSetFieldsOf(campaign))
+            .build();
+
+    // Creates the service client.
+    try (CampaignServiceClient campaignServiceClient =
+        googleAdsClient.getLatestVersion().createCampaignServiceClient()) {
+      // Updates the campaign.
+      MutateCampaignsResponse response =
+          campaignServiceClient.mutateCampaigns(
+              Long.toString(customerId), ImmutableList.of(operation));
+
+      // Displays the results.
+      System.out.printf(
+          "Updated campaign with resource name '%s'.%n", response.getResults(0).getResourceName());
+    }
+  }
+
+  /**
+   * Returns the DSA settings for a campaign. Throws an error if the campaign does not exist or is
+   * not a DSA campaign.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID.
+   * @param campaignId the campaign ID of the campaign to update.
+   * @return the DSA settings for the campaign.
+   */
+  private static DynamicSearchAdsSetting getDsaSetting(
+      GoogleAdsClient googleAdsClient, long customerId, long campaignId) {
+    // Creates the query.
+    // You must request all DSA fields in order to update the DSA settings in the following step.
+    String query =
+        String.format(
+            "SELECT campaign.id, campaign.name, campaign.dynamic_search_ads_setting.domain_name,"
+                + " campaign.dynamic_search_ads_setting.language_code,"
+                + " campaign.dynamic_search_ads_setting.use_supplied_urls_only "
+                + "FROM campaign "
+                + "WHERE campaign.id = '%s'",
+            campaignId);
+
+    // Creates the request.
+    SearchGoogleAdsRequest request =
+        SearchGoogleAdsRequest.newBuilder()
+            .setCustomerId(Long.toString(customerId))
+            .setPageSize(PAGE_SIZE)
+            .setQuery(query)
+            .build();
+
+    // Creates the service client.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Issues the search request.
+      SearchPagedResponse response = googleAdsServiceClient.search(request);
+
+      // Throws an exception if the campaign is not a DSA campaign.
+      // The server will return an error if the campaign is not a DSA campaign, but this
+      // exception is easier to interpret.
+      if (!response
```

Comment:
Formatting looks strange

Reply:
That's correct according to google-java-format.